doh_longest_substring_2.py

```diff
@@ -0,0 +1,33 @@
+
+# coding: utf-8
+
+def longest_substring(s):
+
+    '''
+
+    Arg:
+        s = string to be evaluated for longest substring
+
+    Output:
+        len(substring) = length of the longest substring in int
+
+    '''
+
+    substrings = ''
+
+    for number in range(0, len(set(s))+1)[::-1]:
```

Comment:
The third argument of `range` is the step size. Step size can be negative, which yields a descending range, e.g.
```
In [1]: range(5, 0, -1)
Out[1]: [5, 4, 3, 2, 1]
```
So the preferred way of doing this is:
```
range(len(set(s)), -1, -1)
```
Also `number` is a bad name for a variable!!!